larq/layers.py

```diff
@@ -56,6 +56,14 @@ def build(self, input_shape):
                     values_shape=self.kernel.shape, name=f"{self.name}/flip_ratio"
                 )
 
+    @property
+    def non_trainable_weights(self):
```

Comment:
@koenhelwegen Do you think messing with `non_trainable_weights` could have any unexpected side effects?

I also tried setting `non_trainable_weights = []` on the metrics layer directly, but that doesn't work since initialisation wouldn't be handled correctly.

Reply:
I have a hard time judging if this is save. But I don't really understand why we have to mess with this in the first place. TF Keras metrics are also stateful, so is there not a more standard way of dealing with this?

Reply:
> TF Keras metrics are also stateful, so is there not a more standard way of dealing with this?

TF Keras metrics are added to the model. We add our metrics to the layer since we want to observe changes in the forward pass. Underneath a metric is a normal Keras layer and therefor adding weights will add them to the layer as well.
`layer.weights` internally calls `layers.non_trainable_weights` and `layers.trainable_weights` and will also get all the weights from its child layers (which is important if you have layers that bundle multiple layers). I checked, that the metrics are correctly updated during training in TF 1.14 and TF 2. It seams that we can safely make this change without messing things up.

I'll open an issue on the TensorFlow repo on Monday to see if this behaviour is intended or an oversight. The `add_metrics` API is quite new so there aren't a lot of examples out there that use this on a layer level.

Reply:
okay, thanks! I will merge it and if we find a better way we can change it later